File: mobile/src/main/java/technology/mainthread/apps/gatekeeper/service/MessagingInstanceIdService.java
package technology.mainthread.apps.gatekeeper.service;

import com.google.firebase.iid.FirebaseInstanceIdService;

import javax.inject.Inject;

import dagger.android.AndroidInjection;
import technology.mainthread.apps.gatekeeper.data.RegisterDevices;

public class MessagingInstanceIdService extends FirebaseInstanceIdService {

    @Inject
    RegisterDevices registerDevices;

    @Override
    public void onCreate() {
        AndroidInjection.inject(this);
        super.onCreate();
    }

    @Override
    public void onTokenRefresh() {
        registerDevices.registerDevice();
        startService(RefreshFCMSubscriptionsService.getRefreshFCMSubscriptionsIntent(this));
    }

}

File: mobile/src/main/java/technology/mainthread/apps/gatekeeper/service/MessagingInstanceIdService.kt
package technology.mainthread.apps.gatekeeper.service

import com.google.firebase.iid.FirebaseInstanceIdService
import dagger.android.AndroidInjection
import technology.mainthread.apps.gatekeeper.data.RegisterDevices
import javax.inject.Inject

class MessagingInstanceIdService : FirebaseInstanceIdService() {

    @Inject
    internal lateinit var registerDevices: RegisterDevices

    override fun onCreate() {
        AndroidInjection.inject(this)
        super.onCreate()
    }

    override fun onTokenRefresh() {
        registerDevices.registerDevice()
        startService(RefreshFCMSubscriptionsService.getRefreshFCMSubscriptionsIntent(this))
    }

}
